feature_tests/cpp2/include/AttrOpaque1Renamed.d.hpp

```diff
@@ -9,30 +9,41 @@
 #include <optional>
 #include "diplomat_runtime.hpp"
 #include "AttrOpaque1.d.h"
+#include "CPPRenamedAttrEnum.d.hpp"
 
+class Unnamespaced;namespace ns {
+class AttrOpaque1Renamed;
+class CPPRenamedAttrEnum;
+}
 
+
+namespace ns {
 class AttrOpaque1Renamed {
 public:
 
-  inline static std::unique_ptr<AttrOpaque1Renamed> totally_not_new();
+  inline static std::unique_ptr<ns::AttrOpaque1Renamed> totally_not_new();
```

Comment:
Nit, here and elsewhere: You don't need to restate the namespace when you are inside a `namespace` block. I think how C++ resolution works is it will look in the current namespace and strip one level from the namespace until it either finds the item or fails. So, specifying `ns::Foo` here is potentially ambiguous because if there is a `namespace ns { namespace ns { Foo } }` it will find that one instead of the one at the current level.

Reply:
I disagree: I'd like the generating code to be clean and don't care as much about redundancy in this code. Tracking namespaces everywhere will be somewhat painful

As for ambiguity, my general take is "don't do that", in the long run we can patch this if desired, or use unambiguous `::ns` imports (I _believe_ that works?).